Objects/exceptions.c

```diff
@@ -2295,7 +2295,7 @@ MemoryError_dealloc(PyBaseExceptionObject *self)
 {
     BaseException_clear(self);
 
-    if (Py_TYPE(self) != PyExc_MemoryError) {
+    if (Py_TYPE(self) != (PyTypeObject *)PyExc_MemoryError) {
         return Py_TYPE(self)->tp_free((PyObject *)self);
```

Comment:
I got another compile warning: `D:\a\1\s\Objects\exceptions.c(2324,56): warning C4098: 'MemoryError_dealloc': 'void' function returning a value [D:\a\1\s\PCbuild\pythoncore.vcxproj]` in [here](https://dev.azure.com/Python/cpython/_build/results?buildId=68165&view=logs&j=d554cd63-f8f4-5b2d-871b-33e4ea76e915&t=4e222b70-3701-5c98-1806-ea78a7e04afb&l=234).
Maybe we can cancel `return`?

Reply:
We cannot cancel it, but we can split the expression so the mvsc is happy.